### PR TITLE
Assorted updates to project_update.yml

### PR DESCRIPTION
**Pull request by wenottingham**
1. Use blocks for readability
2. Use variable|default(omit) to clean up some duplicated tasks
3. Add a check that tests whether the repo is already checked out when doing a 'run' job, and short-circuit if so.

For https://github.com/ansible/awx/issues/287

### CODE REVIEW

**Comment by wenottingham** [comment]
Lightly tested with git. Not tested with svn/hg. Requires ansible 2.3+.

File: awx/playbooks/project_update.yml [hunk 1]
```diff
@@ -23,88 +25,111 @@
...
+          force: "{{scm_clean}}"
+          update: false
+          clone: false
+        register: repo_check
+        when: scm_full_checkout|default('')
+        ignore_errors: true
+
+      - name: break if already checked out
+        meta: end_play
+        when: scm_full_checkout|default('') and repo_check|succeeded and repo_check.before == scm_branch
```
Comment (matburt) [inline awx/playbooks/project_update.yml:43]: I'm wondering if this `before == scm_branch` is going to work.

`before` and `after` are hashes and sometimes `scm_branch` is going to be a tag/branch... is this just a shortcut for if we have specified a particular revision?

Comment (wenottingham) [reply awx/playbooks/project_update.yml:43]: AIUI, `scm_branch` for check jobs is a tag/branch, `scm_branch` for run jobs is the revision.

Comment (matburt) [reply awx/playbooks/project_update.yml:43]: ah it is indeed

**Comment by matburt** [comment]
I'm good with this on the surface but I want to verify it against svn and hg before merge.

**Comment by wenottingham** [comment]
Subversion check removed, modules don't support it without connecting to the remote repo, which defeats the point.

**Comment by wenottingham** [comment]
... neither does mercurial (although it looks like it could if the module was better... will see about sending a patch upstream).

Changeset reverted to just do git.

**Review by matburt — APPROVED** [review]

**Pull request merged**